### ISSUE TITLE
Error failed analysis for node #

### ISSUE THREAD

**Issue by RRighart** [sonos/tract#1087]
Using Rust in a Jupyter notebook, I have attempted to load onnx models, that I converted from Keras TF using tf2onnx. I had the following errors: Failed analyse for node #762 "model_dense_1_Softmax" Softmax,
A similar kind of error occurred for another model: Failed analyse for node #51 "sequential_2_dense_6_Sigmoid" Sigmoid.
I used the following code :
```
let model = tract_onnx::onnx()
    .model_for_path("path/to/your/model.onnx")?
    .into_optimized()?
    .into_runnable()?; 

```
Would be glad to have your advice if this is a onnx error or coding error in Rust. Thank you very much in advance!

**Comment by kali** [comment]
Can you cut and paste the full error please ? It is likely you just need to adjust some shapes a bit...

**Comment by RRighart** [comment]
Hi, thank you for your reply. In Jupyter notebook, it did not give any other infos. So I have run it as well in Visual Studio using the following code : 

```
use tract_onnx::prelude::*;

fn main() -> TractResult<()> {
    let model = tract_onnx::onnx()
        .model_for_path("/path-to-EFN-color-new.onnx")?
        .into_optimized()?
        .into_runnable()?;
    Ok(())
}

```
**_Error message:_**

warning: `load-model` (bin "load-model") generated 1 warning (run `cargo fix --bin "load-model"` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 4.37s
     Running `target/debug/load-model`
Error: Failed analyse for node #762 "model_dense_1_Softmax" Softmax

Caused by:
    0: Infering facts
    1: Applying rule inputs[0].shape == outputs[0].shape
    2: Unifying shapes unk__2047,15 and unk__2048,15
    3: Impossible to unify Sym(unk__2047) with Sym(unk__2048).

Thank you very much in advance for your help!

**Comment by kali** [comment]
Try adding `.set_output_fact(0, Default::default())?` before calling `into_optimized()`. Onnx models exports are over-constraint... (equivalent is `--onnx-ignore-output-shapes` on the command line. I'll need to make a FAQ...

**Comment by RRighart** [comment]
The following:
`   .set_output_fact(0, Default::default())?
    .into_optimized()
    .into_runnable()?;`
Gives error[E0599]: no method named `into_optimized` found for unit type `()` in the current scope

**Comment by sCorp-io** [comment]
Was running into the same issue, using `with_output_fact` instead solved it for me
```
.with_output_fact(0, Default::default())?
.into_optimized()
.into_runnable()?;
```

**Comment by RRighart** [comment]
Thank you very much, this works!
